### DISCUSSION TITLE
How to access secrets from my Node.js project code?

### DISCUSSION THREAD

**Discussion by cobrexus** [vercel/vercel#5494] [Help] [answered]
Hi! I'm new to Node.js, and I need to store my database secret/key.

Should I do this with GitHub secrets or Vercel secrets? With either, how do I call the value from my code?

Like with environment variables you would use

```javascript
const databaseKey = process.env.DATABASE_SECRET
```

**Comment by nathanchu** [comment, marked as answer]
As far as I know Vercel cannot access GitHub secrets. If you create a secret on vercel, you can add a secret to a project as an environment variable, so within Serverless Functions, you could use `process.env.DATABASE_SECRET`. See https://vercel.com/docs/environment-variables.

**Reply by cobrexus** [reply]
Thanks for answering.

On the page you linked to, there is this text in the table comparing a plaintext environment variable to a secret environment variable:

![image](https://user-images.githubusercontent.com/34424160/100717972-b2afe800-33b2-11eb-9325-1fe9bc2fd34a.png)

Could you please explain what this means? I can still access the value from my serverless function like how you demonstrated?

**Reply by nathanchu** [reply]
Yes; It just means you can't use it on `vc dev`. But you can use it everywhere else, including serverless functions. The reason this happens is that secrets are encrypted, so no one (including you account) can look at them, and transferring it to your local development environment could reveal it.

**Reply by cobrexus** [reply]
Thanks!
Just a quick question—what's the difference between `vc dev` and `vercel dev`? Or are they the same thing?

**Reply by nathanchu** [reply]
I'm used to running `vc dev` becuase it's shorter, but `vc` is an alias for `vercel`. Either one works, and they are added by default (as long as you have an at least slightly recent version).

**Reply by cobrexus** [reply]
Thank you very much for your help 😄

**Reply by Stoppedwumm** [reply]
Hello, I have the same issue, but when trying the method documented, it says this:

```
Error: secret_key is required
    at new RecaptchaV2 (/var/task/node_modules/express-recaptcha/dist/v2.js:25:19)
    at Object.<anonymous> (/var/task/index.js:12:21)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at /var/task/___now_launcher.js:26:28
    at Object.<anonymous> (/var/task/___now_launcher.js:74:7)
INIT_REPORT Init Duration: 456.48 ms	Phase: invoke	Status: error	Error Type: Runtime.ExitError
Error: Runtime exited with error: exit status 1
```
I have defined it properly and it should work. It seems like the program can't read it

**Reply by Stoppedwumm** [reply]
I fixed it by making it all capital letters
